public/app/plugins/datasource/elasticsearch/language_provider.ts

```diff
@@ -0,0 +1,110 @@
+import { ElasticsearchQuery } from './types';
+import { DataQuery, LanguageProvider } from '@grafana/data';
+
+import { ElasticDatasource } from './datasource';
+
+import { PromQuery } from '../prometheus/types';
+
+const labelRegexp = /(\w+)\s*(=|!=|=~|!~)\s*("[^"]*")/g;
```

Comment:
Seems to work but I would like parsing to be based on the grammar we already have in https://github.com/grafana/grafana/blob/1448767c0822b790823b157a86073f8b3c171645/public/app/plugins/datasource/prometheus/promql.ts#L378

Basically you can do:
```
import Prism from 'prismjs';
import grammar from '../../prometheus/promql'

const tokens = Prism.tokenize(query, grammar);
```
And then iterate the tokens and check their types `label-key`, `label-value` and `operator`. Should make it easier to keep this in sync with any updates to promql and should be also less code. We use this approach around to do suggestions and query manipulation.

Reply:
done